test/fixtures/test-runner/output/hooks-with-no-global-test.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+const { test, describe, it, before, after, beforeEach, afterEach } = require('node:test');
+const assert = require("assert");
+
+// This file should not have any global tests to reproduce bug #48844
+const testArr = [];
+
+before(() => testArr.push('global before'));
+after(() => {
+  testArr.push('global after');
+
+  try {
+    assert.deepStrictEqual(testArr, [
+      'global before',
+      'describe before',
+
+      'describe beforeEach',
+      'describe it 1',
+      'describe afterEach',
+
+      'describe beforeEach',
+      'describe test 2',
+      'describe afterEach',
+
+      'describe nested before',
+
+      'describe beforeEach',
+      'describe nested beforeEach',
+      'describe nested it 1',
+      'describe afterEach',
+      'describe nested afterEach',
+
+      'describe beforeEach',
+      'describe nested beforeEach',
+      'describe nested test 2',
+      'describe afterEach',
+      'describe nested afterEach',
+
+      'describe nested after',
+      'describe after',
+      'global after',
+    ]);
+  } catch (e) {
+    // TODO(rluvaton): remove the try catch after #48867 is fixed
```

Comment:
because of #48867 we must use `process.exit` as otherwise, it won't fail the test